README.md

```diff
@@ -99,18 +115,28 @@ A request for execution should contain the following request parameters:
 * `operationName` - [*Optional*]: The name of the Operation in the Document to execute.
 * `variables` - [*Optional*]: Values for any Variables defined by the Operation.
 
+Note: Be aware that `query` is a misleading name as it can contain a string describing the operation that may be either a query, a mutation or a subscription.
+
 Note: depending on the serialization format used, values of the aforementioned parameters can be
 encoded differently but their names and semantics must stay the same.
 
 Note: specifying `null` in JSON (or equivalent values in other formats) as values for optional request parameters is equivalent to not specifying them at all.
 
 ## GET
 
-For HTTP GET requests, the query parameters should be provided in the query component of the request URL in the form of
+For HTTP GET requests, the query parameters MUST be provided in the query component of the request URL in the form of
 `key=value` pairs with `&` symbol as a separator and both the key and value should have their "reserved" characters percent-encoded as specified in [section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2).
-The unencoded value of the `variables` parameter should be represented as a JSON-encoded string.
+The unencoded value of the `variables` parameter MUST be represented as a URL-encoded JSON string.
+
+GET requests must ONLY be used for executing queries and MUST NOT be used for mutations or subscriptions.
+
+Servers MUST return an _Error response_ if a GET request includes a `query` and `operationName` (if present) describing an operation other than a query, such as a mutation or subscription.
+
+Clients MUST NOT send GET requests for non-query operations.
```

Comment:
Are these lines similar in meaning?

> GET requests must ONLY be used for executing queries and MUST NOT be used for mutations or subscriptions.

and 

> Clients MUST NOT send GET requests for non-query operations.

If not, should we provide an example of a `non-query operation`?